api/src/main/java/io/knotx/server/api/security/AuthHandlerFactory.java

```diff
@@ -38,4 +39,11 @@
    */
   AuthHandler create(Vertx vertx, JsonObject config);
 
+  /**
+   * Register custom routing required by AuthHandler
+   * @param router
+   */
+  default void registerCustomRoute(Router router) {
```

Comment:
Factory registering anything directly on the Router seems like a dangerous idea to me. A factory should not depend on its instances, but I think different instances may have different routes to register.